<?xml version="1.0"?>
<launch>
	<!-- Arguments -->
	<arg name="record" default="false" />
	<arg name="use_sim" default="true" />
	<arg name="use_gripper" default="true" />
	<arg name="start_pose" default="pose_PegInHole4" />
	<!-- <arg name="controller" value="cartesian_impedance_example_controller" /> -->
	<!-- <arg name="controller" value="cartesian_variable_impedance_controller_passive" /> -->
	<arg name="controller" value="cartesian_impedance_controller_damping_ratio" />
	<!-- <arg name="controller" value="cartesian_variable_impedance_controller_damping_ratio" /> -->

	<!-- Start Controller -->
	<!-- Control in Simulation robot -->
	<group if="$(arg use_sim)">
		<include file="$(find franka_gazebo)/launch/sim2.launch" pass_all_args="true">
		<arg name="robot_ip" value="franka" />
		<arg name="arm_id" value="fr3" />
		<arg name="robot" value="fr3" />
		<arg name="use_gripper" value="$(arg use_gripper)" />
		<arg name="controller" value="$(arg controller)" />
		<arg name="start_pose" value="$(arg start_pose)" />
		</include>
	</group>
	<!-- Control real robot -->
	<group unless="$(arg use_sim)">
		<include file="$(find franka_example_controllers)/launch/select_controller.launch" pass_all_args="true">
		<arg name="robot_ip" value="franka" />
		<arg name="arm_id" value="fr3" />
		<arg name="robot" value="fr3" />
		<arg name="load_gripper" value="true" />
		<arg name="launch-prefix" value="sleep 5" />
		<arg name="controller" value="$(arg controller)" />
		<param name="use_sim_time" value="false" />
		</include>
	</group>

	<!-- Run teleop script -->
	<node name="dmp_play" pkg="dmp_ros" type="dmp_play_BuildBot.py" output="screen" required="true" >
        <param name="config_path" value="$(find dmp_ros)/config/dmp_play.yaml"/>
	</node>

	<node pkg="tf2_ros" type="static_transform_publisher" name="FT_frame_broadcaster" args="0 0 0.031 0.7853982 0 0 fr3_EE ft_sensor" />

	<!-- Conditional USB Camera Bringup and Rosbag Record -->
    <group if="$(arg record)">
        <!-- <include file="$(find haptic_ros)/launch/USBCamBringup.launch"/> -->

		<node name="rosbag_record" pkg="rosbag" type="record"
		args="-o $(find dmp_ros)/data/new/relay/
		/franka_state_controller/franka_states
		/franka_state_controller/joint_states
		/franka_state_controller/F_ext
		/franka_state_controller/F_ext_base
		/ft_sensor/ft_compensated
		/ft_sensor/ft_compensated_base
		/franka_gripper/joint_states
		/cartesian_impedance_controller_damping_ratio/equilibrium_pose
		/usb_cam/image_raw/compressed"
		output="screen"/>

    </group>

</launch>
